src/classes/CRLP_ApiExecuteRollups_TEST.cls

```diff
@@ -0,0 +1,300 @@
+     * rollup and the target (summary) record is not updated as a result.
+     */
+    @IsTest
+    private static void shouldCompleteExecuteRollupsForContactSoftCredit() {
+        UTIL_CustomSettingsFacade.getRollupSettingsForTests(new Customizable_Rollup_Settings__c (
+                Customizable_Rollups_Enabled__c = true
+        ));
+
+        CMT_UnitTestData_TEST.mockFullSetOfRollupDefinitions();
+
+        Contact contact1 = UTIL_UnitTestData_TEST.getContact();
+        Contact contact2 = UTIL_UnitTestData_TEST.getContact();
+        insert new List<Contact>{ contact1, contact2 };
+        contact1 = [SELECT Id, AccountId FROM Contact WHERE Id = :contact1.Id LIMIT 1];
+        contact2 = [SELECT Id, AccountId FROM Contact WHERE Id = :contact2.Id LIMIT 1];
+
+        Opportunity opp = new Opportunity(
+                Id = UTIL_UnitTestData_TEST.mockId(Opportunity.SObjectType),
+                AccountId = contact1.AccountId,
+                Amount = 100,
+                CloseDate = Date.today(),
+                StageName = UTIL_UnitTestData_TEST.getClosedWonStage()
+        );
+
+        OpportunityContactRole ocr1 = new OpportunityContactRole(
+                Id = UTIL_UnitTestData_TEST.mockId(OpportunityContactRole.SObjectType),
+                OpportunityId = opp.Id,
+                ContactId = contact1.Id,
+                Role = UTIL_CustomSettingsFacade.DEFAULT_OPPORTUNITY_CONTACT_ROLE_DONOR,
+                IsPrimary = false
+        );
+        OpportunityContactRole ocr2 = new OpportunityContactRole(
+                Id = UTIL_UnitTestData_TEST.mockId(OpportunityContactRole.SObjectType),
+                OpportunityId = opp.Id,
+                ContactId = contact2.Id,
+                Role = UTIL_CustomSettingsFacade.DEFAULT_OPPORTUNITY_CONTACT_ROLE_DONOR,
+                IsPrimary = false
+        );
+
+        Partial_Soft_Credit__c psc = new Partial_Soft_Credit__c(
+                Id = UTIL_UnitTestData_TEST.mockId(Partial_Soft_Credit__c.SObjectType),
+                Opportunity__c = opp.Id,
+                Opportunity__r = opp,
+                Contact__c = contact1.Id,
+                Contact__r = contact1,
+                Role_Name__c = UTIL_CustomSettingsFacade.DEFAULT_OPPORTUNITY_CONTACT_ROLE_DONOR,
+                Contact_Role_Id__c = ocr1.Id,
+                Amount__c = opp.Amount
+        );
+
+        Map<Id, Map<SObjectType, List<SObject>>> rollupData = new Map<Id, Map<SObjectType, List<SObject>>>{
+            opp.Id => new Map<SObjectType, List<SObject>>{
+                Opportunity.SObjectType => new List<SObject>{ opp },
+                OpportunityContactRole.SObjectType => new List<SObject>{ ocr1, ocr2 },
+                Partial_Soft_Credit__c.SObjectType => new List<SObject>{ psc }
+            }
+        };
+
+        Test.startTest();
+
+        CRLP_ApiExecuteRollups service = new CRLP_ApiExecuteRollups()
```

Comment:
This class is unique to processing rollups through the Api, so I think it should keep Api in the name. Without that it could be confused with the `CRLP_RollupProcessor` class. Since there is already a `CRLP_ApiService` class that handles other CRLP related actions via the Api, I feel that `CRLP_ApiExecuteRollups` is still a pretty descriptive name given what it does, even though a good part of that behavior is related to restructuring the data to pass into the actual `CRLP_RollupProcessor` engine.